### PR TITLE
Pass other props to root element

### PR DESCRIPTION
**Pull request by darkowic**
Resolves part of #68

### CODE REVIEW

**Comment by darkowic** [comment]
Contains #70

File: src/Rating.jsx [hunk 1]
```diff
@@ -100,7 +101,8 @@ class Rating extends React.PureComponent {
       direction,
       emptySymbol,
       fullSymbol,
-      placeholderSymbol
+      placeholderSymbol,
+      ...other
```
Comment (mkarajohn) [inline src/Rating.jsx]: Please remove passing other props. Let's open an issue to discuss any specific additions you'd like to see to the API. I don't really like the idea of mass passing of undocumented props.

File: src/Rating.jsx [hunk 2]
```diff
@@ -158,6 +160,7 @@ class Rating extends React.PureComponent {
         style={{ display: 'inline-block', direction }}
         onMouseEnter={!readonly ? this.onMouseEnter : noop}
         onMouseLeave={!readonly ? this.onMouseLeave : noop}
+        {...other}
```
Comment (mkarajohn) [inline src/Rating.jsx]: Please remove passing other props. Let's open an issue to discuss any specific additions you'd like to see to the API. I don't really like the idea of mass passing of undocumented props.

File: src/RatingAPILayer.jsx [hunk 3]
```diff
@@ -70,7 +70,8 @@ class RatingAPILayer extends React.PureComponent {
       fractions,
       direction,
       start,
-      stop
+      stop,
+      ...other
```
Comment (mkarajohn) [inline src/RatingAPILayer.jsx]: Please remove passing other props. Let's open an issue to discuss any specific additions you'd like to see to the API. I don't really like the idea of mass passing of undocumented props.

File: src/RatingAPILayer.jsx [hunk 4]
```diff
@@ -91,6 +92,7 @@ class RatingAPILayer extends React.PureComponent {
         placeholderSymbol={placeholderSymbol}
         onClick={this.handleClick}
         onHover={this.handleHover}
+        {...other}
```
Comment (mkarajohn) [inline src/RatingAPILayer.jsx]: Please remove passing other props. Let's open an issue to discuss any specific additions you'd like to see to the API. I don't really like the idea of mass passing of undocumented props.

**Comment by mkarajohn** [comment]
Apart from comments LGTM 👌

**Comment by dreyescat** [comment]
I have just accepted the pull request #70 having the actual fix for the svg issue #69.

**Comment by dreyescat** [comment]
BTW... I think this propagation breaks the component. It seems that doesn't work anyway. I tried to checkout this branch locally and it fails. I get a bunch of warnings in the console and the rating doesn't get selected.

**Comment by mkarajohn** [comment]
Yeah, I just checked it out as well. Passing `other` without separating all the rest properties that we do not want to pass along with `other` messes things up. 

screen shot 2018-01-05 at 11 44 58

It's not just the warnings, it breaks funcitonality too.

**Comment by darkowic** [comment]
@dreyescat you are right. Sorry about that. I've fixed it

**Comment by darkowic** [comment]
Let's discuss in #68 how we want to do it.

**Comment by RavenHursT** [comment]
Any chance this is gonna get done soon?  I'd love to use this library, but I need to be able to set `className` on the root element as well...

**Comment by dreyescat** [comment]
@RavenHursT According to #68 we agreed to only pass `className` to the wrapper element. This PR doen't seem to be ready yet.

**Comment by dreyescat** [comment]
Solved in #96

**Pull request closed without merging**